src/StreamForwarder.cs

```diff
@@ -0,0 +1,132 @@
+namespace RunScript;
+
+using System.Text;
+
+// https://github.com/dotnet/sdk/blob/a758a468b71e15303198506a8de1040649aa0f35/src/Cli/Microsoft.DotNet.Cli.Utils/StreamForwarder.cs
+public sealed class StreamForwarder
```

Comment:
For now this implementation is fine but ultimately it should be updated so we write directly to the `IConsoleWriter` implementation. This will give us the ability to prefix each line with the task name as well as delay writing until the process is finished which will be needed for parallel execution.